components/AppBar/src/MDCAppBar.m

```diff
@@ -32,6 +33,37 @@
 static NSString *const MDCAppBarHeaderStackViewKey = @"MDCAppBarHeaderStackViewKey";
 static const CGFloat kStatusBarHeight = 20;
 
+@implementation MDCAppBarTextColorAccessibilityMutator
+
+- (nonnull instancetype)init {
+  self = [super init];
+  return self;
+}
+
+- (void)mutate:(nonnull MDCAppBar *)appBar {
+  // Determine what is the appropriate background color
+  // Because navigation bar renders above headerview, it takes presedence
+  UIColor *backgroundColor = appBar.navigationBar.backgroundColor ?:
+      appBar.headerViewController.headerView.backgroundColor;
+  if (!backgroundColor) {
+    return;
+  }
+
+  // Update title label color based on navigationBar/headerView backgroundColor
+  NSMutableDictionary *textAttr =
+      [NSMutableDictionary dictionaryWithDictionary:[appBar.navigationBar titleTextAttributes]];
+  UIColor *textColor = [MDFTextAccessibility textColorOnBackgroundColor:backgroundColor
```

Comment:
in order to get medium is "bold" we need to use the Typography's version of isLargeForContrastRatios.